bugsnag-android-core/src/main/java/com/bugsnag/android/internal/MemoryMappedOutputStream.kt

```diff
@@ -0,0 +1,60 @@
+package com.bugsnag.android.internal
+
+import java.io.File
+import java.io.OutputStream
+import java.io.RandomAccessFile
+import java.nio.MappedByteBuffer
+import java.nio.channels.FileChannel
+
+/**
+ * OutputStream wrapper around a shared memory file.
+ *
+ * Memory-mapped files are maintained by the operating system, and so they are guaranteed to match
+ * the last data written to the memory region, even if the app crashes.
+ *
+ * The memory mapped file is fixed to the length specified at instantiation, meaning that
+ * write() will throw java.nio.BufferOverflowException if the file is too full.
```

Comment:
Could the relevant methods be annotated with [`@Throws`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-throws/) to document this?

Reply:
Added